code/modules/crafting/craft.dm

```diff
@@ -286,15 +298,14 @@
 		Deletion.Cut(Deletion.len)
 		qdel(DL)
 
-
-/datum/personal_crafting/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.not_incapacitated_turf_state)
+/datum/component/personal_crafting/ui_interact(a,b,c,d, mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.not_incapacitated_turf_state)
```

Comment:
Why are you doing this? Just make the signal get proxied through another proc that has the proper arguments. You aren't the only one who's going to be interacting with the code here, make it readable and understandable.

Also you need to add an observer check

Reply:
isobserver checks istype to be dead, component only exists in humans and simple animals with dexterous flag, there is no need to check for observer or any other mob types

Reply:
Observers can see the ui of those they observe